store/tikv/mock-tikv/rpc.go

```diff
@@ -537,3 +418,179 @@ func NewRPCClient(cluster *Cluster, mvccStore *MvccStore) *RPCClient {
+			resp.Commit = &kvrpcpb.CommitResponse{RegionError: err}
+			return resp, nil
+		}
+		resp.Commit = handler.handleKvCommit(r)
+	case tikvrpc.CmdCleanup:
+		r := req.Cleanup
+		if err := handler.checkRequest(r.GetContext(), r.Size()); err != nil {
+			resp.Cleanup = &kvrpcpb.CleanupResponse{RegionError: err}
+			return resp, nil
+		}
+		resp.Cleanup = handler.handleKvCleanup(r)
+	case tikvrpc.CmdBatchGet:
+		r := req.BatchGet
+		if err := handler.checkRequest(r.GetContext(), r.Size()); err != nil {
+			resp.BatchGet = &kvrpcpb.BatchGetResponse{RegionError: err}
+			return resp, nil
+		}
+		resp.BatchGet = handler.handleKvBatchGet(r)
+	case tikvrpc.CmdBatchRollback:
+		r := req.BatchRollback
+		if err := handler.checkRequest(r.GetContext(), r.Size()); err != nil {
+			resp.BatchRollback = &kvrpcpb.BatchRollbackResponse{RegionError: err}
+			return resp, nil
+		}
+		resp.BatchRollback = handler.handleKvBatchRollback(r)
+	case tikvrpc.CmdScanLock:
+		r := req.ScanLock
+		if err := handler.checkRequest(r.GetContext(), r.Size()); err != nil {
+			resp.ScanLock = &kvrpcpb.ScanLockResponse{RegionError: err}
+			return resp, nil
+		}
+		resp.ScanLock = handler.handleKvScanLock(r)
+	case tikvrpc.CmdResolveLock:
+		r := req.ResolveLock
+		if err := handler.checkRequest(r.GetContext(), r.Size()); err != nil {
+			resp.ResolveLock = &kvrpcpb.ResolveLockResponse{RegionError: err}
+			return resp, nil
+		}
+		resp.ResolveLock = handler.handleKvResolveLock(r)
+	case tikvrpc.CmdGC:
+		r := req.GC
+		if err := handler.checkRequest(r.GetContext(), r.Size()); err != nil {
+			resp.GC = &kvrpcpb.GCResponse{RegionError: err}
+			return resp, nil
+		}
+		resp.GC = &kvrpcpb.GCResponse{}
+	case tikvrpc.CmdRawGet:
+		r := req.RawGet
+		if err := handler.checkRequest(r.GetContext(), r.Size()); err != nil {
+			resp.RawGet = &kvrpcpb.RawGetResponse{RegionError: err}
+			return resp, nil
+		}
+		resp.RawGet = handler.handleKvRawGet(r)
+	case tikvrpc.CmdRawPut:
+		r := req.RawPut
+		if err := handler.checkRequest(r.GetContext(), r.Size()); err != nil {
+			resp.RawPut = &kvrpcpb.RawPutResponse{RegionError: err}
+			return resp, nil
+		}
+		resp.RawPut = handler.handleKvRawPut(r)
+	case tikvrpc.CmdRawDelete:
+		r := req.RawDelete
+		if err := handler.checkRequest(r.GetContext(), r.Size()); err != nil {
+			resp.RawDelete = &kvrpcpb.RawDeleteResponse{RegionError: err}
+			return resp, nil
+		}
+		resp.RawDelete = handler.handleKvRawDelete(r)
+	case tikvrpc.CmdCop:
+		r := req.Cop
+		if r.GetTp() == kv.ReqTypeSelect || r.GetTp() == kv.ReqTypeIndex {
+			r.Tp = kv.ReqTypeDAG
+			sel := new(tipb.SelectRequest)
+			err = proto.Unmarshal(r.Data, sel)
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+
+			executors := extractExecutors(sel)
+			dag := &tipb.DAGRequest{
+				StartTs:        sel.GetStartTs(),
+				TimeZoneOffset: sel.TimeZoneOffset,
+				Flags:          sel.Flags,
+				Executors:      executors,
+			}
+			r.Data, err = dag.Marshal()
```

Comment:
r.Data = r.Data[:0]  // reset the Data before reuse it?
By the way, why Cop request proto is different from others ?